packages/shared/src/types/context.ts

```diff
@@ -1,10 +1,84 @@
-export enum ContextTypeEnum {
-  TENANT = 'tenant',
-  APP = 'app',
-}
+import { EnvironmentId } from './environment';
+import { OrganizationId } from './organization';
+
+export type Context = {
+  _id: string;
+  _organizationId: OrganizationId;
+  _environmentId: EnvironmentId;
+
+  id: ContextId;
+  type: ContextType;
+  data: ContextData;
+
+  key: ContextKey;
+
+  createdAt: string;
+  updatedAt: string;
+};
+
+export type ContextType = string;
 
 export type ContextId = string;
 
+export type ContextKey = `${ContextType}:${ContextId}`;
+
+export const createContextKey = (type: ContextType, id: ContextId): ContextKey => `${type}:${id}`;
+
 export type ContextData = Record<string, unknown>;
 
 export const CONTEXT_IDENTIFIER_REGEX = /^[a-zA-Z0-9_-]+$/;
+
+// Context value can be either a simple string id or a rich object
+export type ContextValue =
+  | string
+  | {
+      id: string;
+      data?: ContextData;
+    };
+
+// Context payload is a record of context types to their values
+// Examples:
+// { tenant: "org-acme" } - single key with string value
+// { tenant: "org-acme", app: "jira" } - multi key with string values
+// { tenant: { id: "org-acme", data: { name: "Acme Corp" } } } - single key with rich object
+// { tenant: { id: "org-acme", data: {} }, app: "jira" } - mixed values
+export type ContextPayload = Partial<Record<ContextType, ContextValue>>;
```

Comment:
This "namespace" structure seems maybe a bit better than passing an array of contexts, since the structure is always the same with single and multi context + you don't need extra validation to prevent same types/keys - unless there is an usecase where two context of same types would be needed like `{ tenant: "id-1", tenant: "id-2" }` but that feels like an anti-pattern.